src/plugins/profiling/server/routes/search_flamechart.ts

```diff
@@ -271,71 +271,75 @@ async function queryFlameGraph(
     logger.info('unique downsampled stacktraces: ' + stackTraceEvents.size);
   }
 
-  const resStackTraces = await logExecutionLatency(
+  const nQueries = 4;
+  const results = new Array(nQueries);
+
+  await logExecutionLatency(
     logger,
     'mget query for ' + stackTraceEvents.size + ' stacktraces',
     async () => {
-      if (testing) {
-        return await client.search(
-          {
+      const functions = new Array(nQueries);
+      const promises = new Array(nQueries);
+      const chunkSize = Math.floor(stackTraceEvents.size / nQueries);
+      const stackTraceIDs = [...stackTraceEvents.keys()];
+
+      logger.info('A');
+
+      for (let i = 0; i < nQueries; i++) {
+        functions[i] = async () => {
+          const chunk = stackTraceIDs.slice(chunkSize * i, chunkSize * (i + 1));
+          return client.mget({
             index: 'profiling-stacktraces',
-            size: stackTraceEvents.size,
-            sort: '_doc',
-            query: {
-              ids: {
-                values: [...stackTraceEvents.keys()],
-              },
-            },
-            _source: false,
-            docvalue_fields: ['FrameID', 'Type'],
-          },
-          {
-            querystring: {
-              filter_path: 'hits.hits._id,hits.hits.fields.FrameID,hits.hits.fields.Type',
-            },
-          }
-        );
-      } else {
-        return await client.mget({
-          index: 'profiling-stacktraces',
-          ids: [...stackTraceEvents.keys()],
-          _source_includes: ['FrameID', 'Type'],
-        });
+            ids: [...chunk],
+            _source_includes: ['FrameID', 'Type'],
+          });
+        };
+
+        // Build and send the queries asynchronously.
+        promises[i] = functions[i]();
```

Comment:
The intermediate `functions` array could likely be saved. Not sure how to do that.

Reply:
Are you wanting to save the intermediate values or the functions themselves?

Reply:
No, I don't need the functions (only want to run them).